tests/specs/mcd/word-pack.k

```diff
@@ -22,50 +22,119 @@ endmodule
+            => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48              
+       andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+       andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+       andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+       [simplification]
+
+    rule       UINT48_UINT48 ==Int #WordPackUInt48UInt48(UINT48_1, UINT48_2)
+            => UINT48_1 ==Int maxUInt48 &Int  UINT48_UINT48
+       andBool UINT48_2 ==Int maxUInt48 &Int (UINT48_UINT48 /Int pow48)
+       andBool #rangeUInt(96, UINT48_UINT48)
+       [simplification]
+
+    rule       #WordPackUInt48UInt48(UINT48_1, UINT48_2) ==Int UINT48_UINT48
+            => UINT48_1 ==Int maxUInt48 &Int  UINT48_UINT48
+       andBool UINT48_2 ==Int maxUInt48 &Int (UINT48_UINT48 /Int pow48)
+       andBool #rangeUInt(96, UINT48_UINT48)
+       [simplification]
+
+    rule    ADDR_UINT8 ==Int #WordPackAddrUInt8(ADDR, UINT8)
+            => ADDR  ==Int maxUInt160 &Int  ADDR_UINT8
+       andBool UINT8 ==Int maxUInt8   &Int (ADDR_UINT8 /Int pow160)
+       andBool #rangeUInt(168, ADDR_UINT8) 
+       [simplification]
+
+    rule    #WordPackAddrUInt8(ADDR, UINT8) ==Int ADDR_UINT8
+            => ADDR  ==Int maxUInt160 &Int  ADDR_UINT8
+       andBool UINT8 ==Int maxUInt8   &Int (ADDR_UINT8 /Int pow160)
+       andBool #rangeUInt(168, ADDR_UINT8) 
+       [simplification]
 
 endmodule
 
 module WORD-PACK-HASKELL [kore]
     imports WORD-PACK-COMMON
 
-    rule    ADDR_UINT48_UINT48 ==Int #WordPackAddrUInt48UInt48(ADDR, UINT48_1, UINT48_2)
-         => ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48
-    andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
-    andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
-    andBool #rangeUInt(256, ADDR_UINT48_UINT48)
-      [simplification]
+    rule    { ADDR_UINT48_UINT48:Int #Equals #WordPackAddrUInt48UInt48(ADDR:Int, UINT48_1:Int, UINT48_2:Int) }
+         => { true #Equals (
+                        ADDR   ==Int maxUInt160 &Int  ADDR_UINT48_UINT48              
+              andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+              andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+              andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+              andBool #rangeUInt(160, ADDR)
+            ) }
+            [simplification]
+
+    rule    { #WordPackAddrUInt48UInt48(ADDR:Int, UINT48_1:Int, UINT48_2:Int) #Equals ADDR_UINT48_UINT48:Int }
+         => { true #Equals (
+                        ADDR   ==Int maxUInt160 &Int  ADDR_UINT48_UINT48              
+              andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+              andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+              andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+            ) }
+            [simplification]
```

Comment:
These rules seem like they should be the same since they are just handling two different orientations of `#Equals`; the condition on line 77 doesn't occur in the second rule. And similarly for  lines 106-122. Is it just a matter of us wanting to add things minimally to make the proof succeed?

Reply:
Ah, no no, that should be consistent. Those two rules are just about the two different orientations of equality, line 77 can be removed freely.